sr_edc_ethercat_drivers/src/sr10.cpp

```diff
@@ -0,0 +1,573 @@
+    command->EDC_command = EDC_COMMAND_SENSOR_DATA;
+  }
+  else
+  {
+    command->EDC_command = EDC_COMMAND_CAN_DIRECT_MODE;
+  }
+
+  if (imu_scale_change_)
+  {
+    command->imu_command.command = IMU_COMMAND_SET_SCALE;
+    command->imu_command.argument[0] = imu_scale_acc_;
+    command->imu_command.argument[1] = imu_scale_gyr_;
+    imu_scale_change_ = false;
+  }
+  else
+  {
+    command->imu_command.command = IMU_COMMAND_NONE;
+  }
+
+  // Request for the different sensors' (including tactile) and motors' status information.
+  // It alternates between even and uneven motors everytime.
+  // It also builds the next control command to send to the motors (e.g. torque control)
+  sr_hand_lib->build_command(command);
+
+  ROS_DEBUG("Sending command : Type : 0x%02X ; data : 0x%04X 0x%04X 0x%04X 0x%04X 0x%04X 0x%04X 0x%04X 0x%04X 0x%04X "
+                    "0x%04X 0x%04X 0x%04X 0x%04X 0x%04X 0x%04X 0x%04X 0x%04X 0x%04X 0x%04X 0x%04X",
+            command->to_motor_data_type,
+            command->motor_data[0],
+            command->motor_data[1],
+            command->motor_data[2],
+            command->motor_data[3],
+            command->motor_data[4],
+            command->motor_data[5],
+            command->motor_data[6],
+            command->motor_data[7],
+            command->motor_data[8],
+            command->motor_data[9],
+            command->motor_data[10],
+            command->motor_data[11],
+            command->motor_data[12],
+            command->motor_data[13],
+            command->motor_data[14],
+            command->motor_data[15],
+            command->motor_data[16],
+            command->motor_data[17],
+            command->motor_data[18],
+            command->motor_data[19]);
+
+  build_CAN_message(message);
+}
+
+/** 
+  *  This function handles receiving data from the EtherCAT bus
+  *
+  *  It allows the driver to get the data present on the EtherCAT bus and update its internal structures.
+  *
+  *  It gives us access to the logical memory instantiated during construct().
+  *  We access the data sent by PIC32 in a fashon similar to packCommand() (through direct memeory address).
+  *  
+  *  @param this_buffer The data just being received by EtherCAT
+  *  @param prev_buffer The previous data received by EtherCAT
+  */
+bool SR10::unpackState(unsigned char *this_buffer, unsigned char *prev_buffer)
+{
+  ETHERCAT_DATA_STRUCTURE_0250_PALM_EDC_STATUS *status_data =
+          reinterpret_cast<ETHERCAT_DATA_STRUCTURE_0250_PALM_EDC_STATUS *>(this_buffer + command_size_);
+  ETHERCAT_CAN_BRIDGE_DATA *can_data = reinterpret_cast<ETHERCAT_CAN_BRIDGE_DATA *>(this_buffer + command_size_ +
+                                                                                    ETHERCAT_STATUS_DATA_SIZE);
+  // number of packets received since the start of the driver
+  static unsigned int num_rxed_packets = 0;
+
+  ++num_rxed_packets;
+
+  // Read IMU status and update internal structures
+  readImu(status_data);
```

Comment:
This needs to be moved after the 

```
if (status_data->EDC_command == EDC_COMMAND_INVALID)
{
...
}
```

Reply:
Also here: https://github.com/shadow-robot/sr-ros-interface-ethercat/pull/265/commits/d65c6019a02b359ac64cffaae9b91fe8bea93719

Reply:
Removed for now, as this changes will be part of a separate PR: https://github.com/shadow-robot/sr-ros-interface-ethercat/pull/265/commits/8dffda4fd7ea2bdbc0e483260b80d1107dde9176